### PR TITLE
IPv6 support in backup client

### PR DESCRIPTION
**Pull request by phughk**

### CODE REVIEW

File: community/kernel/src/test/java/org/neo4j/kernel/impl/util/ConvertersTest.java [hunk 1]
```diff
@@ -125,6 +128,53 @@ public void emptyOptionalWhenOnlyPort()
...
+                Pair.of( "2002::/16", new OptionalHostnamePort( "2002::/16", null, null ) ),
+                Pair.of( "[2001::/32]:1234", new OptionalHostnamePort( "2001::/32", 1234, null ) ),
+                Pair.of( "::/96", new OptionalHostnamePort( "::/96", null, null ) ),
+                Pair.of( "[::/96]", new OptionalHostnamePort( "::/96", null, null ) ),
+                Pair.of( "fec0::/10", new OptionalHostnamePort( "fec0::/10", null, null ) ),
+                Pair.of( "[3FFE::/16]", new OptionalHostnamePort( "3FFE::/16", null, null ) ),
+                Pair.of( full, new OptionalHostnamePort( full, null, null ) ),
+                Pair.of( "[" + full + "]", new OptionalHostnamePort( full, null, null ) ),
+                Pair.of( "[" + full + "]:5432", new OptionalHostnamePort( full, 5432, null ) ),
+                Pair.of( "[1::2]:3:4", new OptionalHostnamePort( "1::2", 3, 4 ) ) );
```
Comment (martinfurmanski) [inline community/kernel/src/test/java/org/neo4j/kernel/impl/util/ConvertersTest.java]: Where does this port-range format as specified with a colon in-between come from? In the past I've only seen things like `1.2.3.4:6000-6010`.

File: community/kernel/src/test/java/org/neo4j/kernel/impl/util/ConvertersTest.java [hunk 2]
```diff
@@ -125,6 +128,53 @@ public void emptyOptionalWhenOnlyPort()
         assertFalse( hostname.isPresent() );
     }
 
+    @Test
+    public void ipv6Works()
+    {
+        // with
+        String full = "1234:5678:9abc:def0:1234:5678:9abc:def0";
+        List<Pair<String,OptionalHostnamePort>> cases = Arrays.asList(
+                Pair.of( "::/0", new OptionalHostnamePort( "::/0", null, null ) ),
```
Comment (martinfurmanski) [inline community/kernel/src/test/java/org/neo4j/kernel/impl/util/ConvertersTest.java]: Why are we testing address block notation? Where does the need for this come from?

**Review by martinfurmanski — CHANGES_REQUESTED** [review]
A few questions regarding the content of the PR. Also missing an integration test for the stated functionality of running a backup using IPv6.

We already have some form of IPv6 support in the product afaik, see for example ClusterIpFamilyIT, so parsing of addresses should already exist. Can you reuse that instead?

**Comment by phughk** [comment]
The existing ipv6 processing has been reused by calling HostnamePort from OptionalHostnamePort and then doing some processing on brackets.

**Pull request closed without merging**